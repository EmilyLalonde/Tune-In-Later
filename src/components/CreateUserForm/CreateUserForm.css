.create-user-form {
  display: grid;
  margin-left: 30%;
  margin-top: 5%;
  width: 40%;
}

.create-user-button {
  background-color: #36abdd;
  border: 3px solid #002236;
  border-radius: 6px;
  color: #002236;
  font-size: 30px;
  height: 50px;
}
