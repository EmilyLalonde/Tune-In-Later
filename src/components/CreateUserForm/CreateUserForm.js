import React, { Component } from "react";
import "./CreateUserForm.css";
import PropTypes from 'prop-types'

class CreateUserForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      name: ''
    };
  }

  handleTyping = (e) => {
    this.setState({
      [e.target.name] : e.target.value
    })
  }

  handleSubmit = (e) => {
    e.preventDefault();
    const user = {
      name: this.state.name,
      email: this.state.email,
      password: this.state.password
    }
    this.props.createTheUser(user) 
  }

  render() {
    console.log(this.props)
    return (
      <form className="create-user-form">
        {this.props.error && <p className="error">{this.props.error}</p>}
        <input
          type="text"
          name="name"
          placeholder="Please enter your awesome name"
          value={this.state.name}
          onChange={this.handleTyping}
          className="formInput"
        />
        <input
          type="text"
          name="email"
          placeholder="Please enter email address"
          value={this.state.email}
          onChange={this.handleTyping}
          className="formInput"
        />
        <input
          type="text"
          name="password"
          placeholder="Please enter password"
          value={this.state.password}
          onChange={this.handleTyping}
          className="formInput"
        />
        <button className="create-user-button" onClick={this.handleSubmit}>SUBMIT</button>
      </form>
    );
  }
}

export default CreateUserForm;

CreateUserForm.propTypes = {
  createTheUser: PropTypes.func
}