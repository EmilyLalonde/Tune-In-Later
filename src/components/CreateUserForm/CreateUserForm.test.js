import React from 'react';
import { shallow } from 'enzyme'
import CreateUserForm from './CreateUserForm';

describe('CreateUserForm', () => {
  it('should', () => {
    expect(true).toEqual(true)
  })
})