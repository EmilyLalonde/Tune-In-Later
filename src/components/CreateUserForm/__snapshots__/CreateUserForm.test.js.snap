// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateUserForm should match the snapshot 1`] = `
<form
  className="create-user-form"
>
  <input
    className="formInput"
    name="name"
    onChange={[Function]}
    placeholder="Please enter your awesome name"
    type="text"
    value=""
  />
  <input
    className="formInput"
    name="email"
    onChange={[Function]}
    placeholder="Please enter email address"
    type="text"
    value=""
  />
  <input
    className="formInput"
    name="password"
    onChange={[Function]}
    placeholder="Please enter password"
    type="text"
    value=""
  />
  <button
    className="create-user-button"
    onClick={[Function]}
  >
    SUBMIT
  </button>
</form>
`;
